controllers/rhmi/rhmi_controller.go

```diff
@@ -1698,3 +1717,30 @@ func formatAlerts(alerts []prometheusv1.Alert) (critical resources.AlertMetrics,
 
 	return critical, warning
 }
+
+// function is checking if STS addon parameter Pattern is valid
+// Parameter is Valid only in case:
+// 1.	Parameter exists and value matching AWS Role ARN pattern
+// Parameter is Not valid  in other cases:
+// 2.	parameter exists and value is NOT matching AWS Role ARN pattern
+// 3.	parameter exists and value is empty
+// 4.	parameter does not exists
+func validateAddOnStsRoleArnParameterPattern(client k8sclient.Client, namespace string) (bool, error) {
```

Comment:
no strong opinion but I would rather see this function under resources > sts or some other helper file